pyqtgraph/graphicsItems/PColorMeshItem.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import division
+
+from ..Qt import QtGui, QtCore
+import numpy as np
+from .. import functions as fn
+from .. import debug as debug
+from .GraphicsObject import GraphicsObject
+from ..Point import Point
+from .. import getConfigOption
+from .GradientEditorItem import Gradients
+from ..colormap import ColorMap
+
+try:
+    from collections.abc import Callable
+except ImportError:
+    # fallback for python < 3.3
+    from collections import Callable
+
+__all__ = ['PColorMeshItem']
+
+
+class PColorMeshItem(GraphicsObject):
+    """
+    **Bases:** :class:`GraphicsObject <pyqtgraph.GraphicsObject>`
+
+    TODO
+    """
+
+    sigImageChanged = QtCore.Signal()
+    sigRemoveRequested = QtCore.Signal(object)  # self; emitted when 'remove' is selected from context menu
+
+
+    def __init__(self, x=None, y=None, z=None, cmap='viridis'):
+        """
+        See :func:`setImage <pyqtgraph.ImageItem.setImage>` for all allowed initialization arguments.
+        """
+        GraphicsObject.__init__(self)
+
+        self.x = x
+        self.y = y
+        self.z = z
+        self.qpicture = None  ## rendered image for display
+
+        self.axisOrder = getConfigOption('imageAxisOrder')
+
+        if cmap in list(Gradients.keys()):
```

Comment:
Do you need to convert to a list here?  doing `in` lookups in the `dict.keys()` I know works in python3, does it not work in python2? (added bonus of not converting to list, is that `in` lookups in the hashed objects are significantly faster.